### PR TITLE
Bjg drizzle rtd

### PR DESCRIPTION
**Pull request by braingram**
Another attempt based off of: https://github.com/spacetelescope/drizzle/pull/111
@zacharyburnett @nden @mcara  Would you check if this correctly built all the docs?

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -26,21 +26,52 @@
...
+
+if sys.version_info < (3, 11):
+    import tomli as tomllib
+else:
     import tomllib
-except ImportError:
-    import toml as tomllib
 
-sys.path.insert(1, '..')
+#sys.path.insert(0, os.path.abspath('../'))
```
Comment (zacharyburnett) [inline docs/conf.py:51]: good catch!

**Comment by zacharyburnett** [comment]
The built docs for this PR are hosted at https://spacetelescope-drizzle--112.org.readthedocs.build/en/112/

**Comment by braingram** [comment]
I don't know why the `sys.path` modification was there. It might be possible to `import drizzle` prior to the path modification if the path modification is required for some other reason but I did not try this.

Should I open this PR or does it make more sense to include the change in one of the other PRs?

File: docs/conf.py [hunk 2]
```diff
@@ -126,7 +163,7 @@
 
 # The name for this set of Sphinx documents.  If None, it defaults to
 # "<project> v<release> documentation".
-html_title = '{0} v{1}'.format(project, release)
+html_title = '{project} v{release}'
```
Comment (mcara) [inline docs/conf.py:166]:
```suggestion
html_title = f'{project} v{release}'
```

Comment (mcara) [inline docs/conf.py:166]:
```suggestion
html_title = f'{project} v{release}'
```

File: docs/conf.py [hunk 3]
```diff
@@ -72,21 +103,27 @@
...
+# if len(setup_cfg['authors']) > 3:
+#     author = author + ", et al."
+#
+# copyright = '{0}, {1}'.format(
+#     datetime.datetime.now().year, author)
+# =======
+project = metadata['name']
+author = metadata['authors'][0]['name']
+copyright = '{datetime.datetime.now().year}, {author }'
+#>>>>>>> 2a12fc2... fix `conf.py` to use `pyproject.toml`
```
Comment (mcara) [inline docs/conf.py:119]:
```suggestion
```

File: docs/conf.py [hunk 4]
```diff
@@ -72,21 +103,27 @@
...
+# <<<<<<< HEAD
+# project = setup_cfg['name']
+#
+# author = ', '.join(v['name'] for v in setup_cfg['authors'][:3])
+# if len(setup_cfg['authors']) > 3:
+#     author = author + ", et al."
+#
+# copyright = '{0}, {1}'.format(
+#     datetime.datetime.now().year, author)
+# =======
```
Comment (mcara) [inline docs/conf.py:115]:
```suggestion
```

File: docs/conf.py [hunk 5]
```diff
@@ -72,21 +103,27 @@
...
+project = metadata['name']
+author = metadata['authors'][0]['name']
+copyright = '{datetime.datetime.now().year}, {author }'
```
Comment (mcara) [inline docs/conf.py:118]:
```suggestion
copyright = f'{datetime.datetime.now().year}, {author }'
```

File: docs/conf.py (hunk 2 above)
Comment (zacharyburnett) [reply docs/conf.py:166]: good catch!

**Comment by zacharyburnett** [comment]
> I don't know why the `sys.path` modification was there. It might be possible to `import drizzle` prior to the path modification if the path modification is required for some other reason but I did not try this.
> 
> Should I open this PR or does it make more sense to include the change in one of the other PRs?

you could potentially change the base branch (the branch being merged into) to one on another PR

or alternatively make these changes on another PR (if you have review access I think you can push to that branch on another person's fork)

**Comment by braingram** [comment]
> > I don't know why the `sys.path` modification was there. It might be possible to `import drizzle` prior to the path modification if the path modification is required for some other reason but I did not try this.
> > Should I open this PR or does it make more sense to include the change in one of the other PRs?
> 
> you could potentially change the base branch (the branch being merged into) to one on another PR
> 
> or alternatively make these changes on another PR (if you have review access I think you can push to that branch on another person's fork)

Thanks!

I tried pushing to your `fix_docs` branch and got a "Permission denied" error so what I propose is:
- open a PR from my `bjg-drizzle-rtd` branch to your `fix_docs` branch
- when the docs build there, close this PR and https://github.com/spacetelescope/drizzle/pull/111
- open your `fix_docs` PR https://github.com/spacetelescope/drizzle/pull/98 for review

Does that sound like a good plan?

File: pyproject.toml [hunk 6]
```diff
@@ -1,13 +1,13 @@
 [project]
 name = "drizzle"
 description = "A package for combining dithered images into a single image"
-requires-python = ">=3.8"
+requires-python = ">=3.9"
```
Comment (mcara) [inline pyproject.toml:4]: is this necessary? if so, remove tests for 3.8

Comment (mcara) [reply pyproject.toml]: I would argue that we should not impose requirement to increase the Python version if the code can run in the old version of Python (with an older `numpy` or whatever). Maybe we do not need to run unit tests agains Python 3.8 (so no official support for older versions)?

**Review by mcara — APPROVED** [review]
Thank you!

**Review by mcara — APPROVED** [review]

**Comment by braingram** [comment]
@mcara if it's easier to merge this PR and close #98 and #111 I don't have any objections (I don't have permissions to merge this). I believe the commits captured the contributions from @zacharyburnett and @nden. Thanks all for work fixing this :)

**Comment by zacharyburnett** [comment]
closes #98 and #111

**Pull request merged**